db/db_merge_operator_test.cc

```diff
@@ -0,0 +1,148 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+#include <string>
+#include <vector>
+
+#include "db/db_test_util.h"
+#include "db/forward_iterator.h"
+#include "port/stack_trace.h"
+#include "utilities/merge_operators.h"
+
+namespace rocksdb {
+
+// Test merge operator functionality.
+class DBMergeOperatorTest : public DBTestBase {
+ public:
+  DBMergeOperatorTest() : DBTestBase("/db_merge_operator_test") {}
+
+  void CheckDB(std::vector<std::pair<std::string, std::string>> data) {
+    auto cfd =
+        reinterpret_cast<ColumnFamilyHandleImpl*>(db_->DefaultColumnFamily())
+            ->cfd();
+    auto iter = new ForwardIterator(dbfull(), ReadOptions(), cfd);
+    iter->SeekToFirst();
+    for (auto p : data) {
+      ASSERT_TRUE(iter->Valid());
+      ParsedInternalKey ikey;
+      ASSERT_TRUE(ParseInternalKey(iter->key(), &ikey));
+      ASSERT_EQ(p.first, ikey.user_key);
+      ASSERT_EQ(p.second, iter->value());
+      iter->Next();
+    };
+    ASSERT_FALSE(iter->Valid());
+    delete iter;
+  }
+
+  void CheckIterator(std::vector<std::pair<std::string, std::string>> data) {
+    for (bool iterate_forward : {true, false}) {
+      auto iter = db_->NewIterator(ReadOptions());
+      bool corrupted = false;
+      if (iterate_forward) {
+        iter->SeekToFirst();
+      } else {
+        iter->SeekToLast();
+        std::reverse(data.begin(), data.end());
+      }
+      for (auto p : data) {
+        ASSERT_TRUE(iter->Valid());
+        ASSERT_EQ(p.first, iter->key());
+        if (p.second != "corrupted") {
+          ASSERT_EQ(p.second, iter->value());
+        } else {
+          corrupted = true;
+        }
+        if (!corrupted) {
+          ASSERT_OK(iter->status());
+        } else {
+          ASSERT_TRUE(iter->status().IsCorruption());
+        }
+        if (iterate_forward) {
+          iter->Next();
+        } else {
+          iter->Prev();
+        }
+      }
+      ASSERT_FALSE(iter->Valid());
+      delete iter;
+    }
+  }
+};
```

Comment:
Thanks for implementing this, but we already have this function DBTestBase::VerifyDBFromMap()
you can pass to it a map of k/v and it will verify that the DB have this data.
can we use it instead ?